client/dashboard/app/queries/me-preferences.ts

```diff
@@ -2,31 +2,32 @@ import { fetchPreferences, updatePreferences } from '../../data/me-preferences';
 import { queryClient } from '../query-client';
 import type { UserPreferences } from '../../data/me-preferences';
 
-export const userPreferencesQuery = (
-	preferenceName?: keyof UserPreferences,
-	defaultValue?: UserPreferences[ keyof UserPreferences ]
-) => ( {
+const defaultValues: UserPreferences = {
+	'sites-view': {},
+	'some-string': '',
+};
+
+export const userPreferencesQuery = () => ( {
 	queryKey: [ 'me', 'preferences' ],
 	queryFn: fetchPreferences,
-	select: ( data: UserPreferences ) => {
-		if ( preferenceName ) {
-			return data[ preferenceName ] || defaultValue;
-		}
+} );
 
-		return data;
+export const userPreferenceQuery = < P extends keyof UserPreferences >( preferenceName: P ) => ( {
+	queryKey: userPreferencesQuery().queryKey,
+	queryFn: fetchPreferences,
+	select: ( data: Partial< UserPreferences > ) => {
+		const fetchedValue = data[ preferenceName ];
+		return fetchedValue === undefined ? defaultValues[ preferenceName ] : fetchedValue;
```

Comment:
How about checking whether the data has the `preferenceName` key?

Reply:
I gave this a try but TypeScript doesn't like this one as much. `Partial<T>` can't tell the difference between the key existing or the key being `undefined`. So if you do `preferenceName in data` it is not enough to guarantee to TypeScript that `data[ preferenceName ]` won't be `undefined`.